File: GameServer/MapObjects/MapCircle.cs
using GameServer.Models;
using Newtonsoft.Json;
using System;

namespace GameServer.MapObjects
{
    public class MapCircle : MapObject
    {
        public MapCircle(double x, double y, double radius, TextureEnum texture = TextureEnum.Default)
            : base(x, y, texture)
        {
            Radius = radius;
        }

        public MapCircle(Vector2 pos, double radius, TextureEnum texture = TextureEnum.Default)
            : this(pos.X, pos.Y, radius, texture)
        {
        }

        public MapCircle(MapCircle c)
            : this(c.Position.X, c.Position.Y, c.Radius, c.Texture)
        {
        }

        public double Radius { get; set; }

        [JsonIgnore]
        public double RadiusSquared => Math.Pow(Radius, 2);

        public override string ToString()
        {
            return $"Circle<x:{Position.X} y:{Position.Y}, r:{Radius}>";
        }

        public override object Clone()
        {
            return new MapCircle(this);
        }

        public override bool Equals(MapObject other)
        {
            var c = other as MapCircle;

            if (c == null)
                return false;

            return Position.Equals(c.Position) && Radius == c.Radius;
        }
    }
}

File: GameServer/wwwroot/js/MapObjects/MapCircle.js

function MapCircle(x, y, radius) {

    const that = this;

    that._position = new Vector2(x, y);
    that._radius = radius;

    that.GetX = () => that._position.GetX();
    that.SetX = (value) => that._position.SetX(value);

    that.GetY = () => that._position.GetY();
    that.SetY = (value) => that._position.SetY(value);

    that.GetPosition = () => that._position;
    that.SetPosition = (value) => that._position = value;

    that.GetRadius = () => that._radius;
    that.SetRadius = (value) => that._radius = value;

    that.RadiusSquared = () => Math.pow(that._radius, 2);

    that.Equals = (otherCircle) =>
        that.GetX() === otherCircle.GetX() &&
        that.GetY() === otherCircle.GetY() &&
        that.GetRadius() === otherCircle.GetRadius();

    that.ToString = () => `Circle<x:${that.GetX()} y:${that.GetY()}, r:${that.GetRadius()}>`;
}

MapCircle.Clone = function(mapCircle) {

    return new MapCircle(mapCircle.GetX(), mapCircle.GetY(), mapCircle.GetRadius());
};

// Export module
if (typeof module !== "undefined" && module.hasOwnProperty("exports")) {
    module.exports = MapCircle;
}
